qiskit_experiments/base_experiment.py

```diff
@@ -92,141 +71,207 @@ def __init__(self, qubits, experiment_type=None, circuit_options=None):
+        # generation
+
+    def _transpile(
+        self,
+        circuits: Union[QuantumCircuit, List[QuantumCircuit]],
+        backend: Optional[Backend] = None,
+        **transpile_options,
+    ) -> List[QuantumCircuit]:
+        """Custom transpilation of circuits for running on backend.
+
+        Subclasses may modify this method if they need to customize how
+        transpilation is done, for example to update metadata in the
+        transpiled circuits.
+        """
+        return transpile(circuits, backend=backend, **transpile_options)
 
-    def transpiled_circuits(self, backend=None, **kwargs):
-        """Return a list of experiment circuits.
+    @classmethod
+    def _default_options(cls) -> Options:
+        """Default kwarg options for experiment"""
+        return Options()
+
+    @property
+    def options(self) -> Options:
+        """Return the options for the experiment."""
+        return self._options
+
+    def set_options(self, **fields):
+        """Set the experiment options.
 
         Args:
-            backend (Backend): Optional, a backend object to use as the
-                               argument for the :func:`qiskit.transpile`
-                               function.
-            kwargs: kwarg options for the :meth:`circuits` method, and
-                    :func:`qiskit.transpile` function.
+            fields: The fields to update the options
 
-        Returns:
-            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s.
+        Raises:
+            AttributeError: If the field passed in is not a supported options
+        """
+        for field in fields:
+            if not hasattr(self._options, field):
+                raise AttributeError(
+                    f"Options field {field} is not valid for {type(self).__name__}"
+                )
+        self._options.update_options(**fields)
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        """Default transpiler options for transpilation of circuits"""
+        return Options(optimization_level=0)
+
+    @property
+    def transpile_options(self) -> Options:
+        """Return the transpiler options for the :meth:`run` method."""
+        return self._transpile_options
+
+    def set_transpile_options(self, **fields):
+        """Set the transpiler options for :meth:`run` method.
+
+        Args:
+            fields: The fields to update the options
 
         Raises:
-            QiskitError: if an initial layout is specified in the
-                         kwarg options for transpilation. The initial
-                         layout must be generated from the experiment.
+            QiskitError: if `initial_layout` is one of the fields.
+        """
+        if "initial_layout" in fields:
+            raise QiskitError(
+                "Initial layout cannot be specified as a transpile option"
+                " as it is determined by the experiment physical qubits."
+            )
```

Comment:
Not that I can immediately think of for circuits, maybe for calibration you will have some scheduling specific ones that should not be user changed though?

Reply:
This almost looks like something that the subclasses would like to control. E.g. perhaps some RB experiments would want to run on a high optimization level while calibration experiments would want to avoid this. I suppose subclasses can always override `set_transpile_options`.

Reply:
sounds like we can override `_default_transpile_options` instead.